docs/howto/configure/auth-management.md

```diff
@@ -227,7 +221,97 @@ User account
+   - Example script invocation that creates a CILogon OAuth client for the 2i2c dask-staging hub:
+      ```bash
+      python3 ./deployer/cilogon_auth.py create 2i2c dask-staging daskhub https://dask-staging.2i2c.cloud/hub/oauth_callback
+      ```
+   - If successfull, the script will have created a secret values file under `config/clusters/<cluster_name>/enc-<hub_name>.secret.values.yaml`. This file 
+   holds the encrypted OAuth client id and secret that have been created for this hub.
+   - The unecrypted file contents should look like this:
+      ```yaml
+        jupyterhub:
+          hub:
+            config:
+              CILogonOAuthenticator:
+                client_id: CLIENT_ID
+                client_secret: CLIENT_SECRET
+        ```
+
+2. **Set the hub to _not_ configure Auth0 in the `config/clusters/<cluster_name>/cluster.yaml` file.**
+   To ensure the deployer does not provision and configure an OAuth app from Auth0, the following config should be added to the appropriate hub in the cluster's `cluster.yaml` file.
+
+   ```yaml
+   hubs:
+     - name: <hub_name>
+       auth0:
+         enabled: false
+   ```
+
+3. **If not already present, add the secret hub config file to the list of helm chart values file in `config/clusters<cluster_name>/cluster.yaml`.**
+   If you created the `enc-<hub_name>.secret.values.yaml` file in step 2, add it the the `cluster.yaml` file like so:
+
+   ```yaml
+   ...
+   hubs:
+     - name: <hub_name>
+       ...
+       helm_chart_values_files:
+         - <hub_name>.values.yaml
+         - enc-<hub_name>.secret.values.yaml
+     ...
+   ```
+
+5. **Edit the non-secret config under `config/clusters/<cluster_name>/<hub_name>.values.yaml`.**
+   You should make sure the matching hub config takes one of the following forms.
+
+   ```{warning}
+   When using this method of authentication, make sure to remove the `allowed_users` key from the config.
+   This is because this key will block any user not listed under it **even if** they are valid members of the the organisation or team you are authenticating against.
+
+   Also, the `admin_users` list need to match `allowed_idps` currently.
+   Reference https://github.com/jupyterhub/oauthenticator/issues/494.
+   ```
+
+   To authenticate using CILogon, allowing only a certain identity provider:
+
+    ```yaml
+    jupyterhub:
+      hub:
+        config:
+          JupyterHub:
+            authenticator_class: cilogon
+          CILogonOAuthenticator:
+            oauth_callback_url: https://{{ HUB_DOMAIN }}/hub/oauth_callback
+            username_claim: USERNAME_KEY
+            scope:
+              - openid
+              - email
+              - org.cilogon.userinfo
+            allowed_idps:
+              - 2i2c.org
+              - IDP
+    ```
+  - Check the [CILogon scopes section](https://www.cilogon.org/oidc#h.p_PEQXL8QUjsQm) to checkout available values for `USERNAME_KEY` claim.
```

Comment:
```suggestion
  - Check the [CILogon scopes section](https://www.cilogon.org/oidc#h.p_PEQXL8QUjsQm) to checkout available values for `USERNAME_KEY` claim. 
     Be careful selecting a value, as once picked this can not be easily changed! It should ideally be something
     that the end user *can not change*, as otherwise they can change it after logging in to access other
     users' servers! 
```